### ISSUE TITLE
CUBLAS_WORKSPACE_CONFIG makes torch slow down

### ISSUE THREAD

**Issue by AnticPan** [ultralytics/ultralytics#19127]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

https://github.com/ultralytics/ultralytics/issues/16457 reports the same issue and it was fixed in v8.2.102 https://github.com/ultralytics/ultralytics/pull/16466 by removing line: https://github.com/ultralytics/ultralytics/blob/35f0fde06969ab599379b8f1f5af657be2828dc0/ultralytics/utils/__init__.py#L131
However, since v8.3.2 https://github.com/ultralytics/ultralytics/pull/16583 , the line is added back. It will slow down torch inference speed.

### Environment

```
Ultralytics 8.3.72 🚀 Python-3.10.16 torch-2.5.1 CUDA:0 (NVIDIA A100-SXM4-80GB, 81229MiB)

numpy               ✅ 1.26.4<=2.1.1,>=1.23.0
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.5.1>=1.8.0
torch               ✅ 2.5.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 6.1.1
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.14>=2.0.0
```

### Minimal Reproducible Example

```python
import os
import torch
from ultralytics import YOLO
from diffusers import StableDiffusionXLPipeline

pipe = StableDiffusionXLPipeline.from_pretrained(""stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16).to("cuda")
with torch.inference_mode():
    pipe("test") # warmup
    pipe("test")
    os.environ.pop('CUBLAS_WORKSPACE_CONFIG', None)
    pipe("test")
    os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
    pipe("test")
```

```
100%|█████████████████████████████████████████████████████████| 50/50 [00:06<00:00,  8.26it/s] # warmup
100%|█████████████████████████████████████████████████████████| 50/50 [00:05<00:00,  8.64it/s] # w/ CUBLAS_WORKSPACE_CONFIG
100%|█████████████████████████████████████████████████████████| 50/50 [00:04<00:00, 10.02it/s] # w/o CUBLAS_WORKSPACE_CONFIG
100%|█████████████████████████████████████████████████████████| 50/50 [00:05<00:00,  8.57it/s] # w/ CUBLAS_WORKSPACE_CONFIG
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AnticPan, thank you for bringing this to our attention 🚀! We appreciate your detailed report and reproducible example—it’s super helpful for debugging.

We recommend you first ensure you’re using the latest version of `ultralytics` along with all required dependencies. You can upgrade by running:

```bash
pip install -U ultralytics
```

### Additional Steps 🛠️

If the issue persists:
1. Verify you’re running in a [supported environment](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) with [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).
2. Test the reproducible code snippet on a clean setup (for instance, in free GPU-powered environments like [Colab](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) or [Gradient](https://console.paperspace.com/github/ultralytics/ultralytics)).

### Related CI and Documentation 📚

All [Ultralytics CI tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml) are passing 🟢, ensuring the latest release is verified across macOS, Windows, and Ubuntu. For more about supported [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/), explore our [documentation](https://docs.ultralytics.com/).

---

Our engineers will thoroughly review this issue soon to provide further assistance. Thanks again for contributing to the Ultralytics community! ❤️ For real-time help, feel free to join us on [Discord](https://discord.com/invite/ultralytics) 🎧 or share your findings with others in the [Discourse Community](https://community.ultralytics.com/) or [Subreddit](https://reddit.com/r/Ultralytics).

**Comment by Y-T-G** [comment]
Does this help?

https://github.com/ultralytics/ultralytics/pull/19138

**Comment by Y-T-G** [comment]
`CUBLAS_WORKSPACE_CONFIG` needs to be set before the CUDA context is initialized, otherwise it will keep displaying warnings during training even if you set it right before training. So that's why it's set on import.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #19138](https://github.com/ultralytics/ultralytics/pull/19138). 🎉 This PR, authored by @Y-T-G with contributions from @glenn-jocher, introduces improvements to the handling of deterministic settings during training. Here's a quick overview of the key changes:

### Key Changes:
1. **`unset_deterministic()` Method**: A utility to clean up deterministic training configurations, providing better control over these settings.
2. **CUBLAS Configuration**: Enhanced the model initialization process by setting `CUBLAS_WORKSPACE_CONFIG` for consistent behavior and reduced warnings.
3. **Post-Training Cleanup**: Improved the training process by unsetting deterministic settings once training is complete for better flexibility.
4. **Updated Documentation**: Added information on the new `unset_deterministic()` method to the documentation, making it easier to understand and use.

These improvements aim to enhance flexibility, suppress unnecessary CUDA warnings, and streamline the training workflow. 

### Testing the Fix:
You can test if these changes resolve the issue using one of the following options:
- Install the latest changes directly from the `main` branch:  
  ```bash
  pip install git+https://github.com/ultralytics/ultralytics.git@main
  ```
- Alternatively, wait for the next official release on PyPI.

Feedback is appreciated! Please let us know if the changes in this PR resolve the issue for you, or if you are still experiencing any related problems. 🙏 Thank you for reporting this and contributing to improving the project. Looking forward to hearing how it works for you! 🚀
